data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -215,4 +216,32 @@ public void subscribeToPersistentSubnets(final List<SubnetSubscription> subnetSu
                 .collect(Collectors.toSet());
     validatorApiChannel.subscribeToPersistentSubnets(internalSubnetSubscriptions);
   }
+
+  public SafeFuture<Optional<List<AttesterDuty>>> getAttesterDuties(
+      final UInt64 epoch, final List<Integer> indexes) {
+    if (indexes.isEmpty()) {
+      return SafeFuture.completedFuture(Optional.empty());
```

Comment:
Wouldn't `Optional.empty()` mean the data is unavailable? Should we be returning `Optional.of(emptyList())`? ie we have supplied all the requested data, there just isn't any.

Reply:
We could do that, I had that earlier and changed it when i was testing, but i'll update the test.